### PR TITLE
[#2811] feat(docker): update start up srcipts for Doris Container

### PR DESCRIPTION
**Pull request by zhoukangcn**
### What changes were proposed in this pull request?

There's a very small chance of GitHub CI failure. to avoid this

- add check for BE status
- add retry for adding BE

### Why are the changes needed?

Fix: #2811

### Does this PR introduce _any_ user-facing change?

N/A

### How was this patch tested?

CI

### CODE REVIEW

**Comment by zhoukangcn** [comment]
As mentioned in the #2811 
> It's hard to reproduce in github CI.
> By analyzing the logs of successful runs, I noticed a slight delay in the listening on ports 8030 and 9030.
> Maybe this delay may cause MySQL CLI connect doris FE failed.

Although it can't be reproduced in the github flow, these changes help reduce the probability of the issue occurring. 

Moreover, it print more logs when the next time the issue is detected.

File: dev/docker/doris/start.sh [hunk 1]
```diff
@@ -65,13 +68,56 @@ if [ "$fe_started" = false ]; then
...
+    break
+  else
+    echo "check failed, response: $response"
+  fi
+done
+
+if [ "$be_started" = false ]; then
+  echo "Doris be failed to start"
+
+  cat ${DORIS_BE_HOME}/log/*
```
Comment (yuqi1129) [inline dev/docker/doris/start.sh:93]: By doing `cat ${DORIS_BE_HOME}/log/*`, we can review the log of the backend server?

File: dev/docker/doris/start.sh [hunk 2]
```diff
@@ -65,13 +68,56 @@ if [ "$fe_started" = false ]; then
...
+  cat ${DORIS_FE_HOME}/log/fe.* ${DORIS_BE_HOME}/log/*
 
-if [ $? -ne 0 ]; then
-  echo "Failed to add the BE to the FE"
   exit 1
 fi
 
 # persist the container
-tail -f ${DORIS_FE_HOME}/log/fe.log ${DORIS_BE_HOME}/log/be.log
+tail -f ${DORIS_FE_HOME}/log/fe.log ${DORIS_BE_HOME}/log/be.INFO
```
Comment (qqqttt123) [inline dev/docker/doris/start.sh:123]: be.INFO fe.log? Is it right?

File: dev/docker/doris/start.sh (hunk 1 above)
Comment (zhoukangcn) [reply dev/docker/doris/start.sh:93]: yes, we can cat all backend log to stdout, so Gravitino could print to log file

File: dev/docker/doris/start.sh (hunk 2 above)
Comment (zhoukangcn) [reply dev/docker/doris/start.sh:123]: fe log filename is fe.log
be log filename is be.INFO


image
image

Comment (qqqttt123) [reply dev/docker/doris/start.sh:123]: OK, but it's weird for a system.

**Comment by yuqi1129** [comment]
@jerryshao @qqqttt123 
Do you have any further comments?

**Comment by qqqttt123** [comment]
@zhoukangcn What's the check interval? Is it aligned to Hive container. If we use short interval, it may increase the burden of the container.

**Comment by zhoukangcn** [comment]
@qqqttt123  I don't understand what 'check interval' exactly means. Can you give more information?

**Comment by qqqttt123** [comment]
> @qqqttt123 I don't understand what 'check interval' exactly means. Can you give more information?
> 
> do you mean `sleep 5` is too long?

5s seems ok for me.

**Review by yuqi1129 — APPROVED** [review]
LGTM

**Pull request merged**